includes/Core/Modules/Module_Settings.php

```diff
@@ -96,4 +97,43 @@ function ( $option ) {
 	protected function get_type() {
 		return 'object';
 	}
+
+	/**
+	 * Gets the callback for validating the setting's value.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return callable|null
+	 */
+	protected function get_validate_callback() {
+		return function( $option ) {
+			return $this->validate_required_fields( $option );
```

Comment:
Not every default field is necessarily required to be specified - unless we decide to make that change overall (would probably require some JS changes, so I'd rather not do so). There may be a `Module_Settings` where there are no required fields.

Since this is a callback function its behavior is not as easy to override (you'd need to get the parent's callback and then compose it, including additional `is_wp_error` checks) and would require a bit too many lines of boilerplate for my taste. So I think introducing reusable protected methods for explicitly validating common things (like required fields for now) allows for easier usage, plus it allows a `Module_Settings` where nothing is required to just not do it, rather than having to specify an empty array for required fields.